packages/runtime/client/rest/src/index.ts

```diff
@@ -303,10 +337,34 @@ export function mkRestClient(baseURL: string): RestClient {
         })
       );
     },
-    getContractById(contractId) {
-      return Contract.getContractById(axiosInstance, contractId);
+    getContractById(request) {
+      if (strict) {
+        const result = Contract.GetContractByIdRequest.decode(request);
+        if (result._tag === "Left") {
+          throw new InvalidTypeError(
+            result.left,
+            `Invalid argument to getContractById(${request})`
+          );
+        } else {
+          const test: typeof request = result.right;
+        }
+      }
```

Comment:
I think this pattern could be captured by a function `assertRequest(methodName, guard, request, strict)`

Reply:
Yes, I was thinking the same.